### ISSUE TITLE
Not able to install redisql by all 3 ways

### ISSUE THREAD

**Issue by pawarvijay** [RedBeardLab/rediSQL#43]
macOs Mojave - 10.14.3 - 64bit

**Way 1** - `docker run siscia/redisql:latest` 
**Way 1.1** `docker run -it --net host siscia/redisql`
**Error** - `Can't load module from /usr/local/lib/libredis_sql.so: server aborting`

**Way 2 -** redis-server --loadmodule redisql_v0.9.2_x86_64.so
**Error 1** Can't load module from 4GpEA2y1oztM2V4MmkXpQn.so: server aborting
**Error 2** Can't load module from redisql_v0.9.2_ARMv7.so: server aborting
**Error 3** Can't load module from redisql_v0.9.2_x86_64.so: server aborting

**Way 3** Compiling module ON `cargo build --release` inside cloned repository
**Error** 

**Also Which one i should consider for 64bit ?**
_RediSQL 0.7.0 x86_64 SO
RediSQL 0.7.0 ARMv7 SO
RediSQL 0.7.0 ARMv7 TRIAL SO
RediSQL 0.7.0 x86_64 TRIAL_

**Comment by siscia** [comment]
Hi!

Sorry about that!

Do you mind to re-try now with docker?

`docker run -it --net host siscia/redisql`

Thanks!

**Comment by pawarvijay** [comment]
Thanks for reply 

i think its failing to create container

$ docker run -it --net host siscia/redisql

`1:M 27 Mar 2019 09:40:57.604 # Server initialized
1:M 27 Mar 2019 09:40:57.604 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 27 Mar 2019 09:40:57.605 * Module 'rediSQL' loaded from /usr/local/lib/rediSQL.so
1:M 27 Mar 2019 09:40:57.606 * Ready to accept connections
[2019-03-27T09:40:57Z WARN  telemetrics] Error in making the request to https://telemetrics.redisql.com/v0/statistics: https://telemetrics.redisql.com/v0/statistics: an error occurred trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1924: (unable to get local issuer certificate)
[2019-03-27T09:40:58Z WARN  telemetrics] Error in making the request to https://telemetrics.redisql.redbeardlab.com/v0/statistics: https://telemetrics.redisql.redbeardlab.com/v0/statistics: an error occurred trying to connect: Network is unreachable (os error 101)
[2019-03-27T09:40:58Z WARN  telemetrics] Return error code from http://telemetrics.redisql.redbeardlab.com/v0/statistics
[2019-03-27T09:40:58Z WARN  telemetrics] Warning, impossible to send the telemetrics.
[2019-03-27T09:40:58Z ERROR telemetrics] Telemetrics not reachables, exit!`

`vijays-Air:~ vijay$ docker ps`
`CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES`
`vijays-Air:~ vijay$`

**Comment by siscia** [comment]
No, the 0.9.2 version has telemetrics on, which is a un-documented features (bug in your case) from the moment.

At the moment the plan is to release all the code as open source, even the PRO part, but enable telemetrics so that every hours redisql contact our servers with telemetrics information (complete anonymous data that cannot be used to infer anything about your business, basically what you get from `REDISQL.STATISTICS`).

The 0.9.2 is the worst release ever that simply enable telemetrics without releasing the code as completely open source, but it was a necessary evil.

You got two very quick solutions,
1) Enable your machine to talk with this host: https://telemetrics.redisql.redbeardlab.com (just an https call, port 443)
2) Use the version 0.9.1: `docker run --net host siscia/redisql:0.9.1`

So, to recap.

RediSQL is moving to a different model, everybody get all the features in exchange of a little bit of work from their side to set up correctly the firewall and whatsover is needed to enable the telemetrics, hopefully this will drive companies toward the paid version while leaving hobbistic with a fully featured product!

You get cautch in the worst possible moment of the transiction, where we just enable the telemetrics without a big annoucement or documetation.

**Comment by pawarvijay** [comment]
@siscia thanks for reply it works
but it does not creates any container 
as it is continuously running 
so how will i do `redis-cli` and access redis command line
![Screen Shot 2019-03-28 at 9 51 15 AM](https://user-images.githubusercontent.com/5990947/55129987-34ac3000-513f-11e9-8748-c156ab334350.png)

**Comment by siscia** [comment]
Hi,

redis follows a server-client architecture.

What you are running is a Redis server, now you need a different process to run a client, the `redis-cli`.

In your specific case you can use: 

```docker run --net host siscia/redisql:0.9.1``` 
to start the server and then, in another terminal, 
```docker run --net host -it siscia/redisql:0.9.1 redis-cli``` 
to start a redis-cli. (Please note the `-it` flags passed to the client.)

Let me know if this helps!

Cheers,

**Comment by pawarvijay** [comment]
something is missing , don't know what
![Screen Shot 2019-03-28 at 6 36 36 PM](https://user-images.githubusercontent.com/5990947/55160159-b5dbe500-5188-11e9-8e65-ac315ae2cd7d.png)

**Comment by siscia** [comment]
You need both the server and the client at the same time.

You first start the server:

```docker run --net host siscia/redisql:0.9.1```

then, **keeping the server running**, you start the client as well.

```docker run --net host -it siscia/redisql:0.9.1 redis-cli```

`redis-cli` is an executable that gives you the client, so it need to be installed on your machine or inside the docker container you are running.

Inside `siscia/redisql:0.9.1` is installed, the client started but it wasn't able to connect to the server, indeed it says "Connection refused".

So, to recap:

1. You start the redis server: ```docker run --net host siscia/redisql:0.9.1```
2. Leaving the server running, you start the client 
```docker run --net host -it siscia/redisql:0.9.1 redis-cli```
3. At this point you should have a console working for you!

**Comment by pawarvijay** [comment]
thanks now it works

**Comment by siscia** [comment]
Great!
